cmake/Modules/FindNumPy.cmake

```diff
@@ -39,11 +39,10 @@
 #============================================================================
 
 # Finding NumPy involves calling the Python interpreter
-if(NumPy_FIND_REQUIRED)
-    find_package(PythonInterp REQUIRED)
-else()
-    find_package(PythonInterp)
-endif()
+execute_process(COMMAND "which" "python"
+    RESULT_VARIABLE PYTHONINTERP_FOUND
+    OUTPUT_VARIABLE PYTHON_EXECUTABLE
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
```

Comment:
There is more to `find_package` that finding the Python interpreter. However, we can tell `find_package` where too look by setting `PYTHON_EXECUTABLE`.
This is what we were doing [here](https://github.com/ds4dm/ecole/blob/a838df2ae423b54cd193922d89050d7078b991aa/cmake/Python.cmake#L3). Perhaps we simply need to change that for a `which python` instead of `find_program`?